engine/apps/mattermost/models/channel.py

```diff
@@ -49,11 +49,33 @@ class Meta:
 
     @classmethod
     def get_channel_for_alert_group(cls, alert_group: AlertGroup) -> typing.Optional["MattermostChannel"]:
+        from apps.mattermost.backend import MattermostBackend  # To avoid circular import
+
         default_channel = cls.objects.filter(
             organization=alert_group.channel.organization, is_default_channel=True
         ).first()
 
-        return default_channel
+        if (
+            alert_group.channel_filter is None
+            or not alert_group.channel_filter.notification_backends
+            or not alert_group.channel_filter.notification_backends.get(MattermostBackend.backend_id)
+        ):
+            return default_channel
+
+        channel_id = alert_group.channel_filter.notification_backends[MattermostBackend.backend_id].get("channel")
+        enabled = alert_group.channel_filter.notification_backends[MattermostBackend.backend_id].get("enabled")
+
+        if not enabled or not channel_id:
```

Comment:
Shouldn't we avoid sending a Mattermost notification if the setting is not enabled? So you need to explicitly enable and choose a channel (or leave the channel empty to use default).